Python/ast.c

```diff
@@ -698,6 +698,13 @@ ast_error(struct compiling *c, const node *n, const char *errmsg, ...)
    small_stmt elements is returned.
 */
 
+static string
+new_type_comment(const char *s, struct compiling *c)
+{
+  return PyUnicode_DecodeUTF8(s, strlen(s), NULL);
```

Comment:
The docs for this function say "Return _NULL_ if an exception was raised by the codec." But you never check for `NULL` when using `NEW_TYPE_COMMENT`. I also noticed, few similar call to this function in this file do the null checks (see code for new identifier).

Reply:
Ouch. This was copied from typed_ast. I'll fix it here. (Though I'm not sure when the codec could ever raise an exception -- I guess only when it runs out of memory.)